syntax = "proto3";

option optimize_for = LITE_RUNTIME;

package d_auth;

import "authentication_data.proto";


// Authenticates a UE and/or OTT User present in the local network.
//
// Currently supports 3GPP-style mutual Authentication and Key Agreement (AKA)
// TODO(matt9j) Add ed25519 asymmetric user auth as well
service LocalAuthentication {
    // Request a new 3GPP AKA authorization vector for a particular UE
    rpc GetAuthVector(AKAVectorReq) returns (AKAVectorResp);

    // TODO(matt9j) Work in progress
    rpc ConfirmAuth(AKAConfirmReq) returns (AKAConfirmResp);
}

// TODO(matt9j) Work in progress
message AKAConfirmReq {
    bytes confirmation = 1;
}

// TODO(matt9j) Work in progress
message AKAConfirmResp {
    bytes response = 1;
}
